FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Models/FhirApiOption/ResourceIdSearchOptions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using EnsureThat;
+using Microsoft.Health.Fhir.Synapse.DataClient.Api;
+
+namespace Microsoft.Health.Fhir.Synapse.DataClient.Models.FhirApiOption
+{
+    public class ResourceIdSearchOptions : BaseSearchOptions
+    {
+        public ResourceIdSearchOptions(
+            string resourceType,
+            string resourceId,
+            List<KeyValuePair<string, string>> queryParameters)
+            : base(resourceType, queryParameters)
+        {
+            EnsureArg.IsNotNullOrEmpty(resourceId, nameof(resourceId));
+
+            ResourceId = resourceId;
+
+            QueryParameters.Add(new KeyValuePair<string, string>(FhirApiConstants.IdKey, ResourceId));
+        }
+
+        public string ResourceId { get; set; }
```

Comment:
Will this value get mutated? Or maybe removing the `set` accessor makes more sense.

Reply:
removed.